### PR TITLE
Added ipython-notebook

### PR DESCRIPTION
**Pull request by reinzor**
For analysis of bag data

### CODE REVIEW

**Review by tfoote — CHANGES_REQUESTED** [review]
This is not passing CI due to formatting. Please also add links for the package listed online. We also need at least a debian rule as well.

For guidance see https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md

**Comment by reinzor** [comment]
Added debian rule, cannot really figure out what is wrong with the formatting ..

**Comment by tfoote** [comment]
It looks like the CI ran on the old commit again. Likely part of github's data consistency problems 
image

I've retriggered the build: https://status.github.com/messages

**Comment by tfoote** [comment]
I looked up the packages in the database and noticed that these names are only on older distros:

https://packages.ubuntu.com/search?keywords=ipython-notebook
https://packages.debian.org/search?keywords=ipython-notebook

The project has been renamed to jupyter and has been released as `jupyter-notebook` on all the newer platforms.
https://packages.ubuntu.com/search?keywords=jupyter-notebook
https://packages.debian.org/sid/jupyter-notebook

To that end we should use that key and add backwards compatibility for the older distros like this:
```
jupyter-notebook:
  ubuntu:
    '*': [jupyter-notebook]
    trusty: [ipython-notebook]
    xenial: [ipython-notebook]
```

On debian the legacy platform would similarly be jessie.

**Review by tfoote — APPROVED** [review]
Thanks for the update, lgtm.

**Comment by reinzor** [comment]
Can this be merged?

**Comment by cottsay** [comment]
If this key specifically targeting the executables and not the python library, please add:
```yaml
  fedora: [python3-notebook]
```
per https://apps.fedoraproject.org/packages/python3-notebook

File: rosdep/python.yaml [hunk 1]
```diff
@@ -134,6 +134,14 @@ ipython:
     vivid: [ipython]
     wily: [ipython]
     xenial: [ipython]
+jupyter-notebook:
+  debian:
+    '*': [jupyter-notebook]
+    jessie: [ipython-notebook]
+  ubuntu:
```
Comment (cottsay) [inline rosdep/python.yaml]:
```suggestion
  fedora: [python3-notebook]
  ubuntu:
```

**Review by cottsay — CHANGES_REQUESTED** [review]

**Review by cottsay — APPROVED** [review]

**Pull request merged**